frontend/src/editor/transcription_editor.tsx

```diff
@@ -48,7 +50,40 @@ function renderElement(
             {element.lang}
           </div>
         </div>
-        <div {...attributes} className="grow-1 basis-full" lang={element.lang}>
+        <div
+          {...attributes}
+          className="grow-1 basis-full"
+          lang={element.lang}
+          ref={(x) => {
+            paraElem.current = x;
+            attributes.ref(x);
+          }}
+          onClick={(e) => {
+            if (e.target !== paraElem.current) {
+              return;
+            }
+            const clientX = e.clientX;
+            const childNodes = Array.from(paraElem.current.childNodes) as Element[];
+            let nearest = 0;
+            let nearest_diff = Infinity;
+            for (const [i, node] of childNodes.entries()) {
+              const bbox = node.getBoundingClientRect();
+              let diff;
+              if (clientX < bbox.left) {
+                diff = bbox.left - clientX;
+              } else if (clientX > bbox.right) {
+                diff = clientX - bbox.right;
+              } else {
+                diff = 0;
+              }
+              if (diff < nearest_diff) {
+                nearest_diff = diff;
+                nearest = i;
+              }
+            }
+            window.dispatchEvent(new TextClickEvent(element.children[nearest]));
```

Comment:
can we also remove the handler in 109 following then?

Reply:
Nope, this doesn't work once you have multiline text. maybe we should just handle that cleanly here too and then get rid of 109

Reply:
I would prefer that (I also think, that this might be faster than having many listeners)

Reply:
I rewrote it to rather use the `selection` api and click the correct word

Reply:
Okay, after a bit more investigation: I think it's impossible to perfectly replicate the caret-behaviour in the onclick handler. I instead opted to use the `getSelection`-API in an onclick handler